d3a_api_client/rest_device.py

```diff
@@ -0,0 +1,164 @@
+from d3a_api_client import APIClientInterface
+import os
+import logging
+import requests
+import json
+import ssl
+from functools import wraps
+from concurrent.futures.thread import ThreadPoolExecutor
+from d3a_api_client.websocket_device import WebsocketMessageDispatcher, WebsocketThread
+
+
+root_logger = logging.getLogger()
+root_logger.setLevel(logging.INFO)
+
+
+def logging_decorator(command_name):
+    def decorator(f):
+        @wraps(f)
+        def wrapped(self, *args, **kwargs):
+            logging.debug(f'Sending command {command_name} to device.')
+            return_value = f(self, *args, **kwargs)
+            logging.debug(f'Command {command_name} responded with: {return_value}.')
+            return return_value
+        return wrapped
+    return decorator
+
+
+class RestDeviceClient(APIClientInterface):
+
+    def __init__(self, simulation_id, device_id, domain_name,
+                 websockets_domain_name, is_ssl, autoregister=False):
+        self.simulation_id = simulation_id
+        self.device_id = device_id
+        self.domain_name = domain_name
+        self.retrieve_key_from_server()
+
+        if is_ssl:
+            self.pem_cert = ssl.get_server_certificate(domain_name)
+        self.dispatcher = WebsocketMessageDispatcher(self)
+        self.websocket_thread = WebsocketThread(simulation_id, device_id, self.jwt_token,
+                                                websockets_domain_name, self.dispatcher)
+        self.websocket_thread.start()
+        self.callback_thread = ThreadPoolExecutor(max_workers=5)
+        self.registered = False
+        if autoregister:
+            print(f"Trying to register")
```

Comment:
Please exchange with a logging message. Probably even in the function register().

Reply:
AH, no, i just see your loggoing decorator! Nice! Just delete this line then.

Reply:
Thanks, deleting that